Userland/Libraries/LibWeb/CSS/StyleValues/BasicShapeStyleValue.h

```diff
@@ -7,12 +7,71 @@
 #pragma once
 
 #include <AK/Variant.h>
-#include <LibGfx/DeprecatedPath.h>
 #include <LibWeb/CSS/CSSStyleValue.h>
+#include <LibWeb/CSS/LengthBox.h>
 #include <LibWeb/CSS/PercentageOr.h>
+#include <LibWeb/CSS/StyleValues/PositionStyleValue.h>
 
 namespace Web::CSS {
 
+struct Inset {
+    Gfx::Path to_path(CSSPixelRect reference_box, Layout::Node const&) const;
+    String to_string() const;
+
+    bool operator==(Inset const&) const = default;
+
+    LengthBox inset_box;
+};
+
+struct Xywh {
+    Gfx::Path to_path(CSSPixelRect reference_box, Layout::Node const&) const;
+    String to_string() const;
+
+    bool operator==(Xywh const&) const = default;
+
+    LengthPercentage x;
+    LengthPercentage y;
+    LengthPercentage width;
+    LengthPercentage height;
+};
+
+struct Rect {
+    Gfx::Path to_path(CSSPixelRect reference_box, Layout::Node const&) const;
+    String to_string() const;
+
+    bool operator==(Rect const&) const = default;
+
+    LengthBox box;
+};
+
+struct Circle {
+    enum FitSide {
+        ClosestSide,
+        FarthestSide,
+    };
+
+    using Radius = Variant<LengthPercentage, FitSide>;
```

Comment:
`FitSide` and the `<shape-radius>` type aren't specific to `circle()`, so it probably makes sense to make them top-level instead.